src/System.Private.CoreLib/shared/System/String.cs

```diff
@@ -472,6 +472,13 @@ public static bool IsNullOrWhiteSpace(string value)
             return true;
         }
 
+        /// <summary>
+        /// Returns a reference to the first element of the String. If the string is null, an access will throw a NullReferenceException.
+        /// </summary>
+        [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
+        [NonVersionable]
+        public ref readonly char GetPinnableReference() => ref _firstChar;
+
         internal ref char GetRawStringData() => ref _firstChar;
```

Comment:
Do we still need the internal GetRawStringData, or can we delete it and replace call sites with use of the new method?

Reply:
`GetRawStringData` returns non-readonly ref and most places that use it do depend on it. I do not think changing it to `GetPinnableReference` + `Unsafe.AsRef` would be an improvement.

Reply:
> GetRawStringData returns non-readonly ref

Yup, missed that.